Session9/src/com/itschool/homework/Session9Challenge10.java

```diff
@@ -0,0 +1,25 @@
+package com.itschool.homework;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.util.Scanner;
+
+public class Session9Challenge10 {
+    public static void main(String[] args) {
+        Session9Challenge10 run = new Session9Challenge10();
+        System.out.println(run.isValidDate());
+    }
+
+    public boolean isValidDate() {
+        Scanner input = new Scanner(System.in);
+        System.out.println("Please insert a date in format YYYY-MM-DD");
+        String dateString = input.nextLine();
+
+        try {
+            LocalDate.parse(dateString);
+            return true;
+        } catch (DateTimeException e) {
```

Comment:
nice! However, `LocalDate.parse` says "Throws: DateTimeParseException – if the text cannot be parsed", but you are catching `DateTimeException`, however it is working. Can you explain why are you catching a different exception and why it is working anyway?

Reply:
DateTimeException is a superclass for DateTimeParseException therefore it catches any exceptions of it's subclasses